torch/_inductor/codegen/simd.py

```diff
@@ -620,7 +619,20 @@ def is_compatible(
         except CantSplit:
             return False
 
-    def split_and_set_ranges(self, lengths: List[List[sympy.Expr]]):
+    def split_and_set_ranges(self, lengths: Sequence[Sequence[sympy.Expr]]):
+        groups = [rt.numel for rt in self.range_trees]
+        if not self.inside_reduction:
+            groups[-1] = sympy.S.One
+
+        return self.map_kernel_groups_to_node_sizes(groups, lengths, self.set_ranges)
```

Comment:
Just curious why passing in self.set_ranges as an argument rather than call it in `map_kernel_groups_to_node_sizes` directly? This is kind of a rare pattern. So I'm thinking if there is a common alternative.

Reply:
I want to reuse this code in an upcoming PR where I don't have a Kernel.